app/common_handlers.py

```diff
@@ -366,6 +366,7 @@ async def get(self, request):
         attributes['sub_user_journey'] = sub_user_journey
         attributes['locale'] = locale
         attributes['country_code'] = uprn_ai_return['response']['address']['countryCode']
+        attributes['address_type'] = uprn_ai_return['response']['address']['censusAddressType']
```

Comment:
The values of country_code and address_type are stored in attributes twice, also being stored as countryCode and censusAddressType. Both versions are stored in the session. This arises I believe as we are receiving JSON from the Java service with Camel case, being the Java and Google JSON style guide standard for property names, but python PEP8 recommendation for variables would be snake case. Maybe something to consider later if we end up with more duplicates.

Reply:
Actually the second set of attributes only get sent to the page and not into the session, however I realise that I had duplicated the values for no reason - have removed the python case attributes, and the page now uses the session based JSON versions instead

Reply:
The dictionary "attributes" is a mutable type. I believe with _session['attributes'] = attributes_ both variables are referencing the same mutable python object (dictionary). Later changes such as:
        _attributes['page_title'] = page_title
        attributes['display_region'] = display_region
        attributes['user_journey'] = user_journey
        attributes['sub_user_journey'] = sub_user_journey
        attributes['locale'] = locale
        attributes['country_code'] = uprn_ai_return['response']['address']['countryCode']
        attributes['address_type'] = uprn_ai_return['response']['address']['censusAddressType']_
are placed in the session and carried into the POST request so country code and address type are I believe duplicated under two names in the session.